R/is_date_in_fyyear.R

```diff
@@ -1,17 +1,37 @@
-#' Check if a date in the financial year
+#' Check if a date, or interval is in the financial year
 #'
-#' @description Test to check if a date is within the financial year
+#' @description Test to check if a date, or intervals is within a given
+#' financial year. If given just one date it will check to see if it is in the
+#' given financial year. When supplied with two dates it will check to see if
+#' any part of that date range falls in the given financial year.
 #'
-#' @param date The date to be checked
+#' @param date The main / start date to check
 #' @param year The financial year in the format '1718' as a character
+#' @param date_end (optional) The end date
+#'
+#' @return a logical T/F
 #'
-#' @return logical T/F
 #' @export
 #'
 #' @examples
 #' is_date_in_year(Sys.time(), "2223")
+#' is_date_in_year(
+#'   date = as.Date("2020-01-01"),
+#'   date_end = as.Date("2023-01-01"),
+#'   year = "2122"
+#' )
 #'
 #' @family date functions
-is_date_in_year <- function(date, year) {
-  lubridate::`%within%`(date, fy_interval(year))
+is_date_in_year <- function(date, year, date_end = NULL) {
```

Comment:
```suggestion
is_date_in_fyear <- function(date, year, date_end = NULL) {
```
Name suggestion to match the script

Reply:
I've updated the function name and parameter name as you suggested.

I also re-ordered the parameters as I didn't like having the `year` param in between the two dates but I also wanted the optional param to be last so that you can use the function without having to specify parameter names.